include/GoogleSync/GoogleSync.php

```diff
@@ -46,1088 +53,249 @@
      *
-     * @param \Google_Service_Calendar_Event $event \Google_Service_Calendar_Event Object
-     * @param String $meeting_id SuiteCRM Meeting Id
+     * @param string $id The SuiteCRM user id
      *
      * @return bool Success/Failure
+     * @throws Exception Rethrows if caught from GoogleSyncBase::initUserService
      */
-    public function delEvent($event, $meeting_id)
+    public function doSync($id)
     {
-        // Make sure the calendar service is set up
-        if (!isset($this->gService)) {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'The Google Service is not set up. See setGService Method.');
-            return false;
-        }
-
-        // Make sure we got a meeting_id
-        if (!isset($meeting_id)) {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'This function requires a meeting id as the 2nd parameter');
-            return false;
-        }
-
-        $return = $this->gService->events->delete($this->calendarId, $event->getId());
-
-        // Pull the status code returned to determine Success/Failure
-        $statusCode = $return->getStatusCode();
-
-        if ($statusCode >= 200 && $statusCode <= 299) {
-         // 2xx statusCode = success
-            $GLOBALS['log']->debug(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Received Success Status Code: ' . $statusCode . ' on delete.');
-
-            // This removes the gsync_id reference from the table.
-            $sql = "UPDATE Meetings SET gsync_id = '' WHERE id = '" . $meeting_id . "'";
-            $res = $this->db->query($sql);
-
-            $this->syncedList[] = $meeting_id;
-            return true;
-        } else {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Received Failure Status Code: ' . $statusCode . ' on delete!');
-            return false;
+        try {
+            $this->initUserService($id);
+        } catch (Exception $e) {
+            // TODO: handle the exception instead just loggin it, it will be logged if unhandled by suitecrm core.
+            $this->logger->fatal('Caught exception: ',  $e->getMessage());
+            throw $e;
         }
-    }
 
-    /**
-     * Update SuiteCRM Meeting from Google Calendar Event
-     *
-     * @param Meeting $event_local SuiteCRM Meeting Bean
-     * @param \Google_Service_Calendar_Event $event_remote Google_Service_Calendar_Event Object
-     *
-     * @return Meeting SuiteCRM Meeting Bean
-     */
-    public function updateSuitecrmMeetingEvent($event_local, $event_remote)
-    {
-
-        if ((!isset($event_local) || empty($event_local)) || (!isset($event_remote) || empty($event_remote))) {
-            $GLOBALS['log']->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'ERROR:Missing Variables');
+        $meetings = $this->getUserMeetings($id);
+        if (empty($meetings)) {
+            // TODO: It should be an exception
```

Comment:
@gymad
Oh man... I'm glad you had me look at this.
There is a possibility that $meetings could be empty, if the user doesn't have any SuiteCRM meetings at all.
This wouldn't be an error, and the process should continue, possibly pulling meetings from Google.
I'll have to fix this.

Reply:
Fixed in temp branch